source/agora/network/Manager.d

```diff
@@ -905,6 +908,15 @@ public class NetworkManager
         import std.range;
         import std.typecons;
 
+        // Protect against running multiple times simultaneously
+        synchronized
+        {
+            if (this.isGettingMissingBlockSigs)
+                return;
+            this.isGettingMissingBlockSigs = true;
+        }
```

Comment:
Making it thread safe.

Reply:
Why ? It's not triggered from different thread contexts

Reply:
It can be called from different tasks so I thought this might be safer. I wasn't sure if the `catchupTask` and the `checkNominate` will be the same thread.

Reply:
After checking I see our tasks use `fibers` and not `threads` for asynchronous execution so the thread safety is not a concern. I will remove the thread synchronization.

Reply:
> After checking I see our tasks use `fibers` and not `threads` for asynchronous execution so the thread safety is not a concern. I will remove the thread synchronization.

@hewison-chris Could you let me know where you find our tasks use `fibers`?

Reply:
Here is the vibe.d runTask method code:
https://github.com/bosagora/vibe-core/blob/3e51ff7f3e478539204f05222624457f4ffd7acb/source/vibe/core/core.d#L308-L320
https://github.com/bosagora/vibe-core/blob/3e51ff7f3e478539204f05222624457f4ffd7acb/source/vibe/core/core.d#L396-L438